### ISSUE TITLE
Zimfw doesn't appropriately handle changing to repository fork in zimrc

### ISSUE THREAD

**Issue by fmbarina** [zimfw/zimfw#542]
### Prerequisites

- [X] I've checked the [existing issues](../issues?q=is%3Aissue) and I'm not duplicating a report.
- [X] I've checked the zimfw [Changelog](https://github.com/zimfw/zimfw/blob/master/CHANGELOG.md) and I'm not being affected by documented changes.
- [X] I've checked the zimfw [📢 Announcements](https://github.com/zimfw/zimfw/discussions/426?sort=new) and I'm not being affected by announced changes.

### Bug description

Should a user install a module by adding `zmodule authorA/repo` to zimrc, then switch to a fork (changing the line to `zmodule authorB/repo`) or vice-versa, zim remains stuck with the original repository. This causes errors (as it should) and forces the user to investigate to fix the issue (not ideal). I posit that zim could do better and be more user-friendly in this situation.

### Steps to reproduce

1. ```
   cd ${$(mktemp -d):A} && \
       ZDOTDIR=${PWD} HOME=${PWD} ZIM_HOME=${PWD}/.zim && \
       curl -fsSL https://raw.githubusercontent.com/zimfw/install/master/install.zsh | zsh && \
       exec zsh
   ```
1. ```
   echo 'zmodule fmbarina/evalcache' >>.zimrc
   ```
1. ```
   zimfw install
   ```
1. ```
   sed --in-place 's/fmbarina/mroth/' .zimrc
   ```
1. Try to install the new repository without manually deleting a directory or altering zimrc.
   You can't. I believe there exists no sequence of zimfw commands to do so.

### Current behavior

The `update` and `check` commands fail with:
```
x evalcache: The git URL does not match. Expected https://github.com/mroth/evalcache.git. Will not try to update.
```

The `install` (somewhat mistakenly) reports:
```
Done with install. Restart your terminal for changes to take effect.
```

The `uninstall` command, not that I blame it, also thinks all is well:
```
Done with uninstall.
```

Thus, when encountering this issue, the user is forced to either:

- Revert their changes to zimrc
- Remove the offending zmodule line from their zimrc
- Manually delete the directory of the offending repository

### Expected behavior

I initially assumed zim would either:

- Remove the old repository and clone the new one
  This is the ideal, automagically do everything as the user intends.
  How to actually do it, though, is a discussion involving implementation details.
- Improve error messages
  ```
  x evalcache: The git URL does not match. Expected https://github.com/mroth/evalcache.git. Will not try to update.
  ```
  Maybe more verbosity, or more instructive messages: How should the user proceed? Anything they could look into?

### zimfw info

```shell
zimfw version:        1.14.0 (built at 2024-06-25 17:29:35 UTC, previous commit is 3b7908d)
LANG:                 en_US.UTF-8
LC_TELEPHONE:         pt_BR.UTF-8
LC_MEASUREMENT:       pt_BR.UTF-8
LC_ADDRESS:           pt_BR.UTF-8
LC_NUMERIC:           pt_BR.UTF-8
LC_PAPER:             pt_BR.UTF-8
LC_MONETARY:          pt_BR.UTF-8
LC_TIME:              pt_BR.UTF-8
OSTYPE:               linux-gnu
TERM:                 xterm-256color
TERM_PROGRAM:         WezTerm
TERM_PROGRAM_VERSION: 20240203-110809-5046fc22
ZIM_HOME:             /tmp/tmp.fchUUXflOw/.zim
ZSH_VERSION:          5.9
```

### Additional context

I originally [discussed](https://github.com/zimfw/zimfw/discussions/525) this as a real head-scratcher I had a while back.

N.B. if the user, in order:

1. Removes the zmodule line from their zimrc
1. Runs `zimfw uninstall`
1. Adds the new zmodule line
1. Runs `zimfw install`

Then they won't have this issue.

My belief is that this shouldn't be necessary, but if it is, then it should be clearly stated somewhere in documentation.

From the `README.md`:

>- Added new modules to ~/.zimrc? Run zimfw install.
>- Removed modules from ~/.zimrc? Run zimfw uninstall.
>- Want to update your modules to their latest revisions? Run zimfw update.

Nothing here clearly points towards "if you change the author of a repository you'll have issues so don't do that."

Possible solutions I considered:

- Mention this in documentation/README (may still catch a user off-guard)
- Try handling the author change in zimrc
- Improve error checking/messages
- Add a new zimfw `reinstall`/`reinstall-all`/`purge` command(s).
  Sometimes, it's just easier to nuke an installation and start from scratch.

Assuming this is an issue that needs solving, depending on the chosen solution I could try sending a PR.

**Comment by ericbn** [comment]
@fmbarina your issue, which is more of a feature request, makes sense. Changing the owner of a repository with the same name seems like an edge case, but worth tackling. Currently, the install and uninstall actions just check if the expected directory exists or not, not its contents.

**Comment by fmbarina** [comment]
My bad! Since I was treating it as "a problem in need of solving" I went with bug. I'm glad to hear you think it's worth tackling despite it being an edge case, which I agree with. Very curious to find out what you believe is the best way to go about this.

**Comment by ericbn** [comment]
Released version 1.15.0 with the new `reinstall` action.
